### PR TITLE
Previously LXD profiles were overwritten, now these are merged.

### PR DESCRIPTION
**Pull request by conloos**
##### SUMMARY
Previously LXD profiles were overwritten, now a parameter is added to merge these are merge together.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lxd_profile: Profile Management

##### ADDITIONAL INFORMATION
Previously LXD profiles were overwritten, now, activated by "merge_profile" these are merged.

So far, only configurations that were defined in Ansible have been adopted. Local changes were thus rejected. E.g. a configuration was only made in the config section, then the existing configurations in all other sections disappeared. especially in the default profile that was uncomfortable.

### CODE REVIEW

**Comment by felixfontein** [comment]
Continues #1748.

File: changelogs/fragments/1813-lxd_profile-merge-profiles.yml
Comment (felixfontein) [inline changelogs/fragments/1813-lxd_profile-merge-profiles.yml:2]:
```suggestion
- lxd_profile - added ``merge_profile`` parameter to merge configurations from the play to an existing profile (https://github.com/ansible-collections/community.general/pull/1813). 
```

File: plugins/modules/cloud/lxd/lxd_profile.py
Comment (felixfontein) [inline plugins/modules/cloud/lxd/lxd_profile.py:97]: Please revert this and the above change.

File: plugins/modules/cloud/lxd/lxd_profile.py
Comment (felixfontein) [inline plugins/modules/cloud/lxd/lxd_profile.py:207]: I think you should keep the whitespace above this import, and probably also the one below it.

File: plugins/modules/cloud/lxd/lxd_profile.py
Comment (felixfontein) [inline plugins/modules/cloud/lxd/lxd_profile.py:500]: This and the next change also need to be reverted.

File: changelogs/fragments/1813-lxd_profile-merge-profiles.yml
Comment (conloos) [reply changelogs/fragments/1813-lxd_profile-merge-profiles.yml:2]: Yes

File: plugins/modules/cloud/lxd/lxd_profile.py
Comment (conloos) [reply plugins/modules/cloud/lxd/lxd_profile.py:97]: I will restore the original text. Still, it's wrong. Python does not have a Path type and how the string is to be used can be found in the description.

File: plugins/modules/cloud/lxd/lxd_profile.py
Comment (conloos) [reply plugins/modules/cloud/lxd/lxd_profile.py:207]: Ok

**Review by felixfontein — APPROVED** [review]
Change should be safe (i.e. not break existing usage of the module). Will merge if nobody complains in a few days.

**Comment by felixfontein** [comment]
@conloos thanks for implementing this!

**Comment by Andersson007** [comment]
@conloos hi, thanks for your contribution! We could add your GH login to `.github/BOTMETA.yml` in a corresponding team as a maintainer.
You'll be notified about Issues/PRs related to this stuff and your `shipit` will be counted by bot for automerge (needs two for bugfixes and minor changes).
What do you think?

**Comment by conloos** [comment]
@Andersson007 
Yes, no problem. The corresponding team is the lxd ansible modules?

**Comment by conloos** [comment]
@Andersson007
Thanks ;O)

**Comment by Andersson007** [comment]
@conloos cool, thanks! I found that there's no a special team yet, so I added your login as a maintainer of the module directly (it gives the same rights). Could you please approve the PR https://github.com/ansible-collections/community.general/pull/2310 ?

**Pull request merged**
